### DISCUSSION TITLE
GSM utilities in Global Workflow

### DISCUSSION THREAD

**Discussion by arunchawla-NOAA** [NOAA-EMC/global-workflow#615] [Archive] [answered]
The following scripts and codes are used to create terrain for GSM 

https://github.com/NOAA-EMC/global-workflow/blob/develop/util/ush/ml7_slm30g.sh
https://github.com/NOAA-EMC/global-workflow/tree/develop/util/sorc/terrain.fd
https://github.com/NOAA-EMC/global-workflow/blob/develop/util/ush/terrain.sh

These are part of utility scripts under Global Workflow, which is used for driving GFS/GEFS. Where should these codes reside?

**Comment by WalterKolczynski-NOAA** [comment]
Arun created this discussion while I was composing my response, so I'll repeat the ideas here:

They should probably just be deleted.

If a script is deleted from develop, it shouldn't impact anything in operations one bit. The file would still exist on the necessary branches. While maintaining some backwards compatibility is desirable or even necessary, develop in general should be forward-looking and not concerned with applications that branched off long ago. Otherwise you just keep accumulating cruft, which is part of why global-workflow is in the state it is in today.

global-workflow develop couldn't run GSM right now even if you tried, so terrain.sh is just dead weight. And ufs_utils similarly doesn't seem like an appropriate place, since the UFS to my knowledge only supports the FV3 dycore. If there is somewhere active development is still being done using the GSM (rather than just operational maintenance) that is somehow also still using modern g-w to provide terrain.sh, then maybe the script should move there. But that seems unlikely.

**Comment by arunchawla-NOAA** [comment]
@jack-woollen there are a series of scripts that support creation of orography data for GSM. One of the two systems using the GSM is the CFS. Does the CFS use scripts from the global workflow to create orography data? Or does it use its own version ? We are trying to determine what to do with these scripts.

**Comment by yangfanglin** [comment]
@arunchawla-NOAA  terrain.fd contains the fortran source code as well.  
I have no objection to having this tool kit removed from the workflow as long as it is somehow preserved for future references.   GSM is still widely used in the community around the world.

**Comment by SMoorthi-emc** [comment]
In my opinion, EMC is too shortsighted.
We used to have GSM as a part of UFS, but CMs decided to abandon it.
It is unfortunate that while CFSV2 is still in the operationa, we abandoned
GSM - I had a version in UFS that has all the modern physics but can't use
it
as UFS changed (I still have the code on Dell).
I know Walter is going to say this is not the subject of this discussion,
but this is for the managers - be careful what to keep and what to discard.
(We provided India with GFS, GEFS and CFS and they are using them - it
appears that we abandoned our commitment to them too).
Moorthi

On Thu, Jan 27, 2022 at 11:11 PM Fanglin Yang ***@***.***>
wrote:

> @arunchawla-NOAA <https://github.com/arunchawla-NOAA> terrain.fd contains
> the fortran source code as well.
> I have no objection to having this tool kit removed from the workflow as
> long as it is somehow preserved for future references. GSM is still widely
> used in the community around the world.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/discussions/615#discussioncomment-2063792>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYRPRRO4RXLA7MC5CETUYIJNPANCNFSM5M7PK6OA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> .com>
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

**Comment by arunchawla-NOAA** [comment]
I am fine with archiving this in the correct place. Right now it is folded under global workflow which serves the current UFS development (for global configurations). Is there a repo for the GSM source code ? We can either file this under that, or create a GSM-utils repo. Are there other GSM based utilities that can be moved there ? If we create this who is best served to be the code manager for such a repo. While we are at it, can we improve the documentation ? I had to rely on SME from Fanglin and Moorthi. It will be good to document up front what this is for. Perhaps replace GFS with GSM.

**Comment by arunchawla-NOAA** [comment]
While we are reviewing this, can anyone tell me the use case for overdate.sigma.sh ? The one line comment for this is that it changes the date for a sigma or surface file and relies on using an nwprod util code called overdate.sigma. But it is not used anywhere in the workflow. What is an appropriate location for this script

**Comment by arunchawla-NOAA** [comment]
And the following one as well

https://github.com/NOAA-EMC/global-workflow/blob/develop/util/ush/reduced_gaussian_grid.sh

**Comment by yangfanglin** [comment]
It appears the source code used by reduced_gaussian_grid.sh had already been removed.  I still have a copy of reduced_gaussian_grid.fd and reduced_gaussian_grid.sh (developed by Henry Jung), and the terrain tool kit,  in my local directory.  I will pass them to someone else before I retire :-) if it impossible to find a new home for them.

**Reply by arunchawla-NOAA** [reply]
Fanglin if they are still useful then we can create a repo for them under GSM-utils. Are they useful ?

**Comment by yangfanglin** [comment]
Arun, These are codes that not everyone can write.   It would be a pity to loose them even if they are not actively used for development at the moment.   Creating an GSM-utils to save tools like reduced_gaussian_grid, terrian,  nemsio_cvt,  nemsio_get,  nemsio_read, sfc2gg, ss2gg, cmpu, "old" chgres etc is a good idea.   Another option is to check if the GFS.v14 package is on github, then create a GFSv14 or GSM repo.

**Comment by arunchawla-NOAA** [comment, marked as answer]
OK, We will create a GSM-utils repository that we can then stage these files to. @KateFriedman-NOAA can we create a GSM-Utils repo that can then host these files ? For now please give access to me, @yangfanglin and @SMoorthi-emc so we can move some of these files there

**Reply by KateFriedman-NOAA** [reply]
I have created the requested repo (https://github.com/NOAA-EMC/GSM-Utils) and added @arunchawla-NOAA , @yangfanglin , and @SMoorthi-emc as the initial admins. Feel free to adjust/add access as needed now.
